### PR TITLE
[Feature] Add number of arena trap layers in menu flyout

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes?
- add number of arena trap in menu flyout ie **Spikes (1)**

## Why am I doing these changes?
- closes https://github.com/pagefaultgames/pokerogue/issues/2038

## What did change?
- Added tag count for arena trap tags

### Screenshots/Videos
- Tag count appear just like the remaining turns for moves like Tailwind
Screenshot 2024-06-12 at 12 21 34 AM

- If arena trap has only 1 max layers, it wont show the count
Screenshot 2024-06-12 at 12 18 55 AM

- Edge case, simultaneous Arena Trap Tag (Spikes)
Screenshot 2024-06-12 at 12 10 39 AM

- Edge case, simultaneous Arena Tag used (Tailwind)
tailwind

- Clearing arena tags still work

https://github.com/pagefaultgames/pokerogue/assets/68144167/33fec8c7-d3f7-4685-ae6d-6f4b8dda019f

- Tests passed
Screenshot 2024-06-12 at 12 37 15 AM

## How to test the changes?
- set the moves of your mon to ArenaTrapTags like Stealth Rock, Spikes or Toxic Spikes
- set moves that are ArenaTags but is not ArenaTrapTag like Tailwind or Trick Room
- give your mon Defog to clear Arena trap tags

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
- [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
- [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/ui/arena-flyout.ts [hunk 1]
```diff
@@ -257,19 +259,42 @@ export default class ArenaFlyout extends Phaser.GameObjects.Container {
...
-          ? ArenaEffectType.FIELD
-          : tagAddedEvent.arenaTagSide === ArenaTagSide.PLAYER
-            ? ArenaEffectType.PLAYER
-            : ArenaEffectType.ENEMY,
+        name,
+        effecType: arenaEffectType,
         maxDuration: tagAddedEvent.duration,
-        duration: tagAddedEvent.duration});
+        duration: tagAddedEvent.duration,
+        tagType: tagAddedEvent.arenaTagType
```
Comment (torranx) [inline src/ui/arena-flyout.ts:291]: tagType is added here to be the identifier since name is being updated to include the count of layers

**Review by bennybroseph — APPROVED** [review]
